lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -53,20 +53,19 @@ def objects
 
     def collection_ancestor? collections
       collections.each do |check|
-        return true if check.id == self.id
         return true if ancestor?(check)
       end
       false
     end
 
     def ancestor? collection
-      return true if collection.id == self.id
+      return true if collection == self
       return false if collection.child_collections.empty?
       current_collections = collection.child_collections
       next_batch = []
       while !current_collections.empty? do
         current_collections.each do |c|
-          return true if c.id == self.id
+          return true if c == self
```

Comment:
When the objects aren't saved, the id=nil.  So the objects can be different but appear the same as nil==nil returns true.  jcoyne suggested object test instead to avoid this issue.